gdg/__init__.py

```diff
@@ -166,7 +177,16 @@ def find_image(name):
         images = [get_relative_path(baseurl, i.path) for i in Image.select().where(Image.path.regexp(pattern)).order_by(SQL('path collate nocase'))]
     
     if len(images) > 1:
-        images.sort(key=lambda x: filename_lev(name.lower(), x.lower()))
+        if not 'api' in cherrypy.request.app.config:
+            return "You haven't configured the goddamn API."
+
+        max_dist = cherrypy.request.app.config['api']['max_lev_distance']
+        max_dist = -1 if max_dist is None else max_dist
+
+        # Filter and then sort the images by their levenshtein distance
+        filtered_images = filter_images_by_lev(name, images, max_dist)
+        filtered_images.sort(key=lambda tup: tup[1])
```

Comment:
I usually prefer using named key-value pairs rather than bare tuples for things like this, maybe something like `{ 'image': filename, 'distance': lev_dist }` (if you can think of better keys, that's cool too)